Fixing tests and Docs

TornaxO7 opened the pull request:
- Provided more docs
- Refactored tests and added more tests

TornaxO7 committed:
Fixing tests and Docs

- Provided more docs
- Refactored tests and added more tests

TornaxO7 commented:
@soywod  Review pls

PS: You were right, we don't need another `Test` trait. I created the needed default-functions in the test modules. :+1:

soywod commented on src/config/model.rs:
```diff
@@ -367,16 +353,60 @@ impl Config {
         let name = account.name.as_ref().unwrap_or(&self.name);
+
+        let special_chars: Vec<&str> = name
+            .matches(|character| ",.:;".contains(character))
```
Where did you get this information? I tried to search into the [RFC 5322](https://datatracker.ietf.org/doc/html/rfc5322) but I did not find those chars.

TornaxO7 replied:
I remembered that you should quote the name, if it includes some special characters. I just remembered those. Now after a quick lookup I found [this](https://datatracker.ietf.org/doc/html/rfc2822#section-3.2.1). Looks like that we'll have to expand the list :D

soywod commented on src/config/model.rs:
```diff
@@ -367,16 +353,60 @@ impl Config {
             format!("{}", account.email)
+        } else if !special_chars.is_empty() {
+            // so the name has special characters => Wrap it with '"'
+            format!("\"{}\" <{}>", name, account.email)
```
Maybe the simpler would be to always quote the display name, what do you think? See https://stackoverflow.com/a/15606313/6213048.

TornaxO7 replied:
Hm... I'm a little bit unsure about this because if we take a look [on the examples](https://datatracker.ietf.org/doc/html/rfc2822#appendix-A) then we can see that the names are quoted, if they include these preserved characters (like the dot or comma). Personally I'd prefer to use the quotes if they're necessary but feel free to decide. I accept quotes as ^^

soywod replied:
I just try to find a good compromise between value of feature and time spent, we see how complex and full of details those RFC are. Putting quotes avoid invalid names. Putting quotes around a name that does not need them seems not to make it invalid either. Knowing that, do you think it's worth spending time digging the RFC to have a perfect working model (with tests) rather than just putting quotes ? I'm open to discussion ofc, I may be wrong there, it is just my opinion.

TornaxO7 replied:
So you'd prefer to have always quotes?

soywod replied:
As you wish. I don't want to force anything. After all, if it's just chars to check then it's okey, you can leave this way.

TornaxO7 committed:
Expanding specials

Added more "special characters" which will add some quotes around the name if it
includes at least one of them.

soywod commented on src/config/model.rs:
```diff
@@ -367,16 +353,60 @@ impl Config {
     pub fn address(&self, account: &Account) -> String {
         let name = account.name.as_ref().unwrap_or(&self.name);
+
+        let special_chars: Vec<&str> = name
```
I think the algorithm can be improved here. We don't need to collect _all_ special chars, we just need to know if there is _at least_ one. What about sth like `let has_special_chars = name.contains(|c| "()<>[]:;@.,".contains(c));`?

TornaxO7 committed:
Fixing test

Improved the detection if the mail-name includes a special character or not.

TornaxO7 committed:
Variable renaming

Renamed a variable for better readability.

TornaxO7 commented:
@soywod update!
